packages/ui/src/app/pages/Proposals/ProposalPreview.tsx

```diff
@@ -98,7 +101,10 @@ export const ProposalPreview = () => {
       />
     )
   }
-
+  if (!proposal) {
+    if (!isLoading) history.replace('/404')
+    return <Loading />
+  }
```

Comment:
If `proposal` is `undefined` it will never get to this line. Instead on line `90` there could be a:
```tsx
if (!proposal || !votes) {
  if (!proposal && !isLoading) {
    history.replace('/404')
  }
  return (
    <PageLayout
    ...
```